packages/storage/src/providers/s3/types/outputs.ts

```diff
@@ -25,52 +26,119 @@ export interface ItemBase {
+/**
+ * type for S3 list item.
+ */
+export type ListOutputItem = Omit<ItemWithKey, 'metadata'>;
 
 /**
  * Output type for S3 downloadData API.
+ * @deprecated Use {@link DownloadDataOutputWithPath} instead.
  */
 export type DownloadDataOutput = DownloadTask<
-	StorageDownloadDataOutput<ItemWithKeyAndPath>
+	StorageDownloadDataOutput<ItemWithKey>
 >;
-
 /**
- * Output type for S3 uploadData API.
+ * Output type with path for S3 downloadData API.
  */
-export type UploadDataOutput = UploadTask<ItemWithKeyAndPath>;
+export type DownloadDataWithPathOutput = DownloadTask<
+	StorageDownloadDataOutput<ItemWithPath>
+>;
 
 /**
  * Output type for S3 getUrl API.
+ * @deprecated Use {@link GetUrlWithPathOutput} instead.
  */
 export type GetUrlOutput = StorageGetUrlOutput;
+/**
+ *  Output type with path for S3 getUrl API.
+ * */
+export type GetUrlWithPathOutput = StorageGetUrlOutput;
 
 /**
- * Output type for S3 getProperties API.
+ * Output type for S3 uploadData API.
+ *  @deprecated Use {@link UploadDataOutputWithPath} instead.
  */
-export type GetPropertiesOutput = ItemWithKeyAndPath;
+export type UploadDataOutput = UploadTask<ItemWithKey>;
+/**
+ *  Output type with path for S3 uploadData API.
+ * */
+export type UploadDataWithPathOutput = UploadTask<ItemWithPath>;
 
 /**
- * Output type for S3 Copy API.
- */
-export type CopyOutput = Pick<ItemWithKeyAndPath, 'key' | 'path'>;
+ * Output type for S3 getProperties API.
+ * @deprecated Use {@link GetPropertiesOutputWithPath} instead.
+ * */
+export type GetPropertiesOutput = ItemBase & StorageItemWithKey;
+/**
+ *  Output type with path for S3 getProperties API.
+ * */
+export type GetPropertiesWithPathOutput = ItemBase & StorageItemWithPath;
 
 /**
- * Output type for S3 remove API.
+ * @deprecated Use {@link ListAllOutputWithPath} instead.
+ * Output type for S3 list API. Lists all bucket objects.
  */
-export type RemoveOutput = Pick<ItemWithKeyAndPath, 'key' | 'path'>;
+export type ListAllOutput = StorageListOutput<ListOutputItemWithKey>;
 
 /**
- * Output type for S3 list API. Lists all bucket objects.
+ * Output type with path for S3 list API. Lists all bucket objects.
  */
-export type ListAllOutput = StorageListOutput<ListOutputItem>;
+export type ListAllWithPathOutput = StorageListOutput<ListOutputItemWithPath>;
 
 /**
+ * @deprecated Use {@link ListPaginateOutputWithPath} instead.
  * Output type for S3 list API. Lists bucket objects with pagination.
  */
-export type ListPaginateOutput = StorageListOutput<ListOutputItem> & {
+export type ListPaginateOutput = StorageListOutput<ListOutputItemWithKey> & {
 	nextToken?: string;
 };
+
+/**
+ * Output type with path for S3 list API. Lists bucket objects with pagination.
+ */
+export type ListPaginateWithPathOutput =
+	StorageListOutput<ListOutputItemWithPath> & {
+		nextToken?: string;
+	};
+
+/**
+ *  Output type with path for S3 copy API.
+ * @deprecated Use {@link CopyOutputWithPath} instead.
```

Comment:
Need to change this to `CopyWithPathOutput`
```suggestion
 * @deprecated Use {@link CopyWithPathOutput} instead.
```